ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/PaywallOptions.kt

```diff
@@ -43,6 +43,13 @@ data class PaywallOptions internal constructor(
         dismissRequest = builder.dismissRequest,
     )
 
+    internal val dataHash: String = run {
```

Comment:
Note that I'm not overriding the `hashCode`/`equals` functions here since we are comparing them (including the listeners) for other methods, like the `updateOptions` method. This was the least disruptive approach that didn't mean recreating the view model for every recreation of the listeners.

Reply:
that makes sense, but let's add a comment so future our future selves understand why we have `dataHash` in the first place